verification/sign.go

```diff
@@ -0,0 +1,189 @@
+		t.Errorf("[ERROR]: Signature varies for same label, want %v but got %v", signedData.HmacOrSignatureR, signedDataWithSameLabel.HmacOrSignatureR)
+	}
+
+	// Rerun with different label, signature must change this time
+	newLabel := make([]byte, digestLen)
+	for i := range newLabel {
+		newLabel[i] = byte(0)
+	}
+
+	signedDataWithDiffLabel, err := c.Sign(handle, newLabel, SignFlags(IsSymmetric), tbs)
+	if err != nil {
+		t.Fatalf("[FATAL]: Error while signing %v", err)
+	}
+
+	if bytes.Equal(signedDataWithDiffLabel.HmacOrSignatureR, signedData.HmacOrSignatureR) {
+		t.Errorf("[ERROR]: Signature must vary for different label despite having same toBeSigned content, want new signature but got old %v", signedData.HmacOrSignatureR)
+	}
+}
+
+// Checks whether the digital signature returned by Sign command can be verified
+// using public key in signing key certificate returned by CertifyKey command.
+// Inspite of the DPE profile supporting symmetric key, for symmetric signing it must be enabled
+// explicitly in Sign command. Else asymmetric signing is used as default.
+func testAsymmetricSigning(d TestDPEInstance, c DPEClient, t *testing.T, handle *ContextHandle, digestLen int) {
+	var err error
+	flags := SignFlags(0)
+
+	seqLabel := make([]byte, digestLen)
+	for i := range seqLabel {
+		seqLabel[i] = byte(i)
+	}
+
+	tbs := make([]byte, digestLen)
+	for i := range tbs {
+		tbs[i] = byte(i)
+	}
+
+	signResp, err := c.Sign(handle, seqLabel, flags, tbs)
+	if err != nil {
+		t.Fatalf("[FATAL]: Error while signing %v", err)
+	}
+
+	// Get signing key certificate using CertifyKey command
+	certifiedKey, err := c.CertifyKey(handle, seqLabel, CertifyKeyX509, CertifyKeyFlags(0))
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not CertifyKey: %v", err)
+	}
+
+	// Check certificate structure
+	var leafCert *x509.Certificate
+	if leafCert, err = x509.ParseCertificate(certifiedKey.Certificate); err != nil {
+		t.Fatalf("[FATAL]: Could not parse certificate using crypto/x509: %v", err)
+	}
+
+	// Read public key
+	var ec elliptic.Curve
+	x := new(big.Int).SetBytes(certifiedKey.Pub.X)
+	y := new(big.Int).SetBytes(certifiedKey.Pub.Y)
+
+	if digestLen == 32 {
+		ec = elliptic.P256()
+	} else if digestLen == 48 {
+		ec = elliptic.P384()
+	}
+
+	publicKey := ecdsa.PublicKey{Curve: ec, X: x, Y: y}
+
+	// Build Signature from bytes
+	r := new(big.Int).SetBytes(signResp.HmacOrSignatureR)
+	s := new(big.Int).SetBytes(signResp.SignatureS)
+
+	// Verify Signature
+	valid := ecdsa.Verify(&publicKey, tbs, r, s)
+	if !valid {
+		t.Error("Signature Verification failed")
+	}
+
+	// Validate certificate chain validation establish the CA of signing key certificate
+	certChainBytes, err := c.GetCertificateChain()
+	if err != nil {
+		t.Fatalf("[FATAL]: Could not get Certificate Chain: %v", err)
+	}
+	certChain := checkCertificateChain(t, certChainBytes)
+	validateLeafCertChain(t, certChain, leafCert)
```

Comment:
nit: Probably not necessary. This is already tested in the CertifyKey test

Reply:
Ok, removed